### PR TITLE
[Dashboard] Authentication

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?
Client Authentication Implementation.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/dashboard/metrics_exporter/api.py [hunk 1]
```diff
@@ -15,20 +15,24 @@ def authentication_request(url, cluster_id) -> AuthResponse:
     auth_requeset = AuthRequest(cluster_id=cluster_id)
     response = requests.post(url, data=auth_requeset.json())
     response.raise_for_status()
-    return AuthResponse.parse_obj(response.json())
+    return AuthResponse.parse_obj(json.loads(response.json()))
```
Comment (simon-mo) [inline python/ray/dashboard/metrics_exporter/api.py:18]: Something similar to pydantic parse_raw? https://pydantic-docs.helpmanual.io/usage/models/#model-properties

File: python/ray/dashboard/metrics_exporter/client.py [hunk 2]
```diff
@@ -25,6 +26,7 @@ class MetricsExportClient:
 
     def __init__(self, address, dashboard_controller, dashboard_id, exporter):
         self.dashboard_id = dashboard_id
+        self.address = address
```
Comment (simon-mo) [inline python/ray/dashboard/metrics_exporter/client.py:29]: Assert check the web protocol?

File: python/ray/dashboard/metrics_exporter/schema.py [hunk 3]
```diff
@@ -59,7 +59,8 @@ def json(self):
     @classmethod
     def parse_obj(cls, obj):
         # Validation.
-        assert type(obj) == dict, ("It can only parse dict type object.")
+        assert type(obj) == dict, ("It can only parse dict type object, "
+                                   "but {} type is given.".format(type(obj)))
```
Comment (simon-mo) [inline python/ray/dashboard/metrics_exporter/schema.py:63]: Nice! Informative error message is good.

**Review by simon-mo — APPROVED** [review]
Looks good!

File: python/ray/dashboard/metrics_exporter/client.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/dashboard/metrics_exporter/client.py:29]: Yeah that's a good idea.

File: python/ray/dashboard/metrics_exporter/api.py (hunk 1 above)
Comment (rkooo567) [reply python/ray/dashboard/metrics_exporter/api.py:18]: What about `parse_json_string`?

Comment (rkooo567) [reply python/ray/dashboard/metrics_exporter/api.py:18]: Will just keep it as is for now.

File: python/ray/dashboard/metrics_exporter/client.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/dashboard/metrics_exporter/client.py:29]: I think it is not that important for now. I will add a check once it is exposed to users.

**Pull request merged**